ansible/roles/mambaforge/tasks/mambaforge.yml

```diff
@@ -25,19 +17,37 @@
   failed_when: false
   register: conda_version_cmd
 
+- name: Wipe the existing installation if it differs from the target version
+  ansible.builtin.file:
+    path: "{{ mamba_home }}"
+    state: absent
+  when:
+    - conda_version_cmd['rc'] is defined
+    - conda_version_cmd['stdout'] is defined
+    - (conda_version_cmd['rc'] == 0) and (conda_version_cmd['stdout'] != 'conda ' ~ mambaforge_conda_version)
+
+- name: Verify that the installation directory doesn't exist
+  ansible.builtin.stat:
+    path: "{{ mamba_home }}"
+  register: mambaforge_home_check
+  failed_when: mambaforge_home_check['stat']['exists']
+
+- name: Create the installation directory
+  ansible.builtin.file:
+    path: "{{ mamba_home }}"
+    state: directory
+    owner: root
+    group: root
+    mode: "0755"
+
 - name: Execute the installation script
   ansible.builtin.command:
     argv:
       - /tmp/mambaforge.sh
       - -b
-      - -u
       - -p
       - "{{ mamba_home }}"
   changed_when: true
-  when:
-    - conda_version_cmd['rc'] is defined
-    - conda_version_cmd['stdout'] is defined
-    - (conda_version_cmd['rc'] != 0) or (conda_version_cmd['stdout'] != 'conda ' ~ mambaforge_conda_version)
```

Comment:
We'd need this condition to still be applied here too, right?

Reply:
I removed it here because at this stage we've already passed the check "Verify that the installation directory doesn't exist".

Reply:
But this now doesn't cover the case for which no action is needed, i.e., valid Conda installation and `conda` version matches, right?

AFAICT, the PR's current state is

| conda version | wanted action | fulfilled? |
|---|---|---|
| unknown | fail task | Yes, due to "Verify that the installation directory doesn't exist" failing |
| matches | no action | NO, due to "Wipe..." condition not fulfilled and "Verify... doesn't exist" failing |
| differs | remove dir and reinstall | Yes, due to "Wipe..." condition fulfilled and "Verify... doesn't exist" passes |

Reply:
You're right! Let me rework this.